linera-client/src/chain_listener.rs

```diff
@@ -57,75 +57,62 @@ pub struct ChainListenerConfig {
     pub delay_after_ms: u64,
 }
 
-#[async_trait]
-pub trait ClientContext {
-    type ValidatorNodeProvider: LocalValidatorNodeProvider;
-    type Storage: Storage;
+type ContextChainClient<C> =
+    ChainClient<<C as ClientContext>::ValidatorNodeProvider, <C as ClientContext>::Storage>;
+
+#[cfg_attr(not(web), async_trait, trait_variant::make(Send))]
```

Comment:
It is like that on `main`, too: https://github.com/linera-io/linera-protocol/blob/4c60efb406c547151aa531103fad2885b91fd420/linera-client/src/chain_listener.rs#L63
And it actually doesn't build without it:

```
error: future cannot be sent between threads safely
   --> linera-client/src/chain_listener.rs:135:13
    |
135 | /             async move {
136 | |                 if let Err(err) =
137 | |                     Self::run_client_stream(chain_id, context, storage, config, listening).await
138 | |                 {
...   |
141 | |             }
142 | |             .in_current_span(),
    | |______________________________^ future created by async block is not `Send`
    |
note: captured value is not `Send`
   --> linera-client/src/chain_listener.rs:137:55
    |
137 |                     Self::run_client_stream(chain_id, context, storage, config, listening).await
    |                                                       ^^^^^^^ has type `Arc<futures::lock::Mutex<C>>` which is not `Send`
note: required by a bound in `linera_base::task::spawn`
   --> /mnt/data/git/linera-protocol/linera-base/src/task.rs:22:44
    |
22  |     pub fn spawn<F: Future<Output: Send> + Send + 'static>(
    |                                            ^^^^ required by this bound in `spawn`
help: consider further restricting this bound
    |
132 |         C: ClientContext + std::marker::Send,
    |                          +++++++++++++++++++

error: could not compile `linera-client` (lib) due to 1 previous error
```

Reply:
Ah it's probably because we have `-> impl Future` blocks, which:
- `trait-variant` requires for functions with default implementations
- `async-trait` doesn't touch.
It might be better to use just plain old
```rust
#[allow(async_fn_in_trait)]
#[cfg_attr(not(web), trait_variant::make(Send))]
```
rather than sometimes `async-trait` and sometimes `trait-variant`, but not high priority.

Reply:
Yes, let's fix that on `main`; no need to backport, probably.